extmod/asyncio/stream.py

```diff
@@ -100,20 +109,40 @@ def drain(self):
 # Create a TCP stream connection to a remote host
 #
 # async
-def open_connection(host, port):
+async def open_connection(host, port, ssl=None, server_hostname=None):
     from errno import EINPROGRESS
     import socket
 
     ai = socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM)[0]  # TODO this is blocking!
     s = socket.socket(ai[0], ai[1], ai[2])
-    s.setblocking(False)
-    ss = Stream(s)
+    if not ssl:
+        s.setblocking(False)
+        ss = Stream(s)
     try:
         s.connect(ai[-1])
```

Comment:
For ssl connections, this will be a blocking connect, which is bad.  The stream should be set to non-blocking for both ssl and non-ssl connections.